CondTools/Hcal/interface/CmdLine.h

```diff
@@ -0,0 +1,396 @@
+//
+//         /* Switches that do not require subsequent arguments */
+//         b = cmdline.has("-b");
+//
+//         /* Declare the end of option processing. Unconsumed options  */
+//         /* will cause an exception to be thrown.                     */
+//         cmdline.optend();
+//
+//         /* Process all remaining arguments */ 
+//         while (cmdline)
+//         {
+//             std::string s;
+//             cmdline >> s;
+//             positionalArgs.push_back(s);
+//         }
+//     }
+//     catch (const CmdLineError& e) {
+//         std::cerr << "Error in " << cmdline.progname() << ": "
+//                   << e.str() << std::endl;
+//         return 1;
+//     }
+//
+//     /* ..... Do main processing here ..... */
+//
+//     return 0;
+// }    
+//
+// Short version options must use a single character. It is possible
+// to combine several short options together on the command line,
+// for example, "-xzvf" is equivalent to "-x -z -v -f".
+//
+// Use standalone "-" to indicate that the next argument is either
+// an option value or the program argument (but not a switch), even
+// if it starts with "-". This is useful if the option value has to
+// be set to a negative number.
+//
+// Use standalone "--" (not preceded by standalone "-") to indicate
+// the end of options. All remaining arguments will be treated as
+// program arguments, even if they start with "-".
+//
+// Note that each of the parsing methods of the CmdLine class ("option",
+// "has", and "require") is greedy. These methods will consume all
+// corresponding options or switches and will set the result to the last
+// option value seen. It is therefore impossible to provide a collection
+// of values by using an option more than once. This is done in order to
+// avoid difficulties with deciding what to do when multiple option values
+// were consumed by the user code only partially.
+//
+// After the "optend()" call, the "argc()" method of the CmdLine 
+// class can be used to determine the number of remaining program
+// arguments. If the expected number of arguments is known in advance,
+// the simplest way to get the arguments out is like this (should be
+// inside the "try" block):
+//
+// if (cmdline.argc() != n_expected)
+//     throw CmdLineError("wrong number of command line arguments");
+// cmdline >> arg0 >> arg1 >> ...;
+//
+// I. Volobouev
+// January 2016
+//=========================================================================
+
+#ifndef CMDLINE_HH_
+#define CMDLINE_HH_
+
+#include <list>
+#include <sstream>
+#include <cstring>
+#include <utility>
+#include <cstdio>
+
+#ifdef __GNUC__
+#include <cstdlib>
+#include <typeinfo>
+#include <cxxabi.h>
+#endif
+
+#ifdef __GXX_EXPERIMENTAL_CXX0X__
+#include <memory>
+#define CmdLine_shared_ptr std::shared_ptr
+#else
+#include <tr1/memory>
+#define CmdLine_shared_ptr std::tr1::shared_ptr
+#endif
+
+// Subsequent classes will throw exceptions of the following class
+class CmdLineError
```

Comment:
such generic class names should be in a name space (or be prepended with some TLA)